File: treatment/src/main/java/com/hartwig/actin/treatment/database/TrialConfigDatabase.java
package com.hartwig.actin.treatment.database;

import java.util.List;

import com.hartwig.actin.treatment.database.config.CohortDefinitionConfig;
import com.hartwig.actin.treatment.database.config.InclusionCriteriaConfig;
import com.hartwig.actin.treatment.database.config.TrialDefinitionConfig;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class TrialConfigDatabase {

    @NotNull
    public abstract List<TrialDefinitionConfig> trialDefinitionConfigs();

    @NotNull
    public abstract List<CohortDefinitionConfig> cohortDefinitionConfigs();

    @NotNull
    public abstract List<InclusionCriteriaConfig> inclusionCriteriaConfigs();
}

File: treatment/src/main/kotlin/com/hartwig/actin/treatment/trial/TrialConfigDatabase.kt
package com.hartwig.actin.treatment.trial

import com.hartwig.actin.treatment.trial.config.CohortDefinitionConfig
import com.hartwig.actin.treatment.trial.config.InclusionCriteriaConfig
import com.hartwig.actin.treatment.trial.config.InclusionCriteriaReferenceConfig
import com.hartwig.actin.treatment.trial.config.TrialDefinitionConfig

data class TrialConfigDatabase(
    val trialDefinitionConfigs: List<TrialDefinitionConfig>,
    val cohortDefinitionConfigs: List<CohortDefinitionConfig>,
    val inclusionCriteriaConfigs: List<InclusionCriteriaConfig>,
    val inclusionCriteriaReferenceConfigs: List<InclusionCriteriaReferenceConfig>
)